Add section on environment write access (#7842)
dbt-labs/docs.getdbt.com: website/docs/docs/cloud/manage-access/about-access.md

:::

> ### Group access and permissions

Reviewer: I wonder if the Intro paragraph (line 24 - I can't access it in this PR) can be rewritten to account for this new info:
> Managing user access in dbt involves a combination of licenses, groups, permissions, and role-based access control (RBAC). Before diving into the specific ways you can assign and regulate access, it’s helpful to understand how dbt defines users and groups, since these form the foundation for all access and permission settings.

Author: This is covered by RBAC (which Brian Jan also wrote an awesome blog post about). The docs were just a little behind here

> You can control which projects the members of any given group have access to with the **Access and permissions** settings. Select the role (or roles) you want to assign to the group, the projects to which they'll have access, and which environments they can write to (edit). 

Reviewer: ```suggestion
Groups in dbt let you manage access and permissions for multiple users at once. Instead of assigning permissions user by user, you can organize people into groups and give them the right level of access based on their role or responsibilities.
```

Author: This is established at the top level H2, so maybe we can jump right in here. How about
```suggestion
The **Access & Permissions** section of a group is where you can assign users the right level of access based on their role or responsibilities. You decide:
```

and then lead into the bullets below

> This gives you the flexibility to determine exactly how much access users in any given group will have. You can provide members with administrative access to their own project, but limit their view of all others to `read-only`. Grant users write access only to the environments they require, while keeping production off limits. 

Reviewer: ```suggestion
When you configure group permissions using **Access and permissions** settings, you decide:
* Projects the group can access
* Roles the group members can perform (for example, admin, developer, or read-only)
* Environments the group can edit

This setup gives you fine-grained control. For example, you might allow one group of analysts to edit jobs in their project, but only let them view production runs elsewhere. Or you could grant admin-level access for a team that owns a specific project while keeping others restricted to read-only.
```

Author: I think we should avoid "fine grained control" because that's a very real thing in group settings, what we offer is only a part of the pie (for example we don't allow customers to pick individual permissions for a group, which is an oft requested feature). How about: 
```suggestion

* Projects the group can access
* Roles that the group members are assigned for each
* Environments the group can edit

This setup provides you with the flexibility to determine the level of access users in any given group will have. For example, you might allow one group of analysts to edit jobs in their project, but only let them view related projects, or you could grant admin-level access to a team that owns a specific project while keeping others restricted to read-only.
```

<Lightbox src="/img/docs/dbt-cloud/dbt-cloud-enterprise/access-control/sample-access-policy.png" width="60%" title="Assign a variety of roles and access permissions to user groups." />

#### Environment write access

> Some permission sets have read-only access to environment settings but are unable to edit them. When you create a group and assign **Environment write access**, it will override any read-only environment settings inherent to the permission set. 

Reviewer: ```suggestion
Some permission sets have read-only access to environment settings, which means they are unable to edit these settings. If you want group members to create, edit, or delete jobs in environments, you must explicitly grant **Environment write access**. This setting overrides the default read-only permissions in a permission set.
```

Author: I think that second sentence could be confusing - it reads like they need to assign it to the group if they want _any_ member to be able to write (and it's not explicitly just for jobs). How about:
```suggestion
Some permission sets have read-only access to environment settings, which means they cannot edit these settings. If you assign users with these permission sets to a group with **Environment write access**, they will be able to create, edit, and delete environment settings, overriding the read-only restriction.
```

Reviewer: Much clearer!!

> In this example, the `analyst` permission set, which by default has read-only access to jobs, is assigned to the group across all projects; however, the **Environment write access** is set to `All Environments`. This will grant all users in this group the ability to create, edit, and delete jobs across all environments and projects. 

Reviewer: ```suggestion
In the following example, the `analyst` permission set, which by default has read-only access to jobs, is assigned to the group across all projects; however, the **Environment write access** is set to `All Environments`.  This grants all users in this group the ability to create, edit, and delete jobs across all environments and projects. 
```

<Lightbox src="/img/docs/dbt-cloud/dbt-cloud-enterprise/access-control/environment-write.png" width="60%" title="Users assigned environment write access will be able to edit environment settings." />

> It's essential to set the **Environment write access** settings only if you intend to grant users the ability to edit them. To grant users only the permissions inherent to their set, leave this setting blank (all boxes unchecked).

Reviewer: I feel like the UI should call this out too?
```suggestion
Only use **Environment write access** settings when you intend to grant users the ability to edit environments. To grant users only the permissions inherent to their set, leave this setting blank (all boxes unchecked).
```

:::

### Group access and permissions

The **Access & Permissions** section of a group is where you can assign users the right level of access based on their role or responsibilities. You decide:

* Projects the group can access
* Roles that the group members are assigned for each
* Environments the group can edit

This setup provides you with the flexibility to determine the level of access users in any given group will have. For example, you might allow one group of analysts to edit jobs in their project, but only let them view related projects, or you could grant admin-level access to a team that owns a specific project while keeping others restricted to read-only.

<Lightbox src="/img/docs/dbt-cloud/dbt-cloud-enterprise/access-control/sample-access-policy.png" width="60%" title="Assign a variety of roles and access permissions to user groups." />

#### Environment write access

> Some permission sets grant users read-only access to environment settings that can be overridden if you assign them to a group with *Environment write access*. They will then be able to create, edit, and delete environment settings, bypassing the read-only restriction.

Author: ```suggestion
Some permission sets grant users read-only access to environment settings that can be overridden if you assign them to a group with **Environment write access**. They will then be able to create, edit, and delete environment settings, bypassing the read-only restriction.
```

> Some permission sets have read-only access to environment settings, which means they cannot edit these settings. If you assign users with these permission sets to a group with **Environment write access**, they will be able to create, edit, and delete environment settings, overriding the read-only restriction.

Reviewer: ```suggestion
Some permission sets grant users read-only access to environment settings that can be overridden if you assign them to a group with *Environment write access*. They will then be able to create, edit, and delete environment settings, bypassing the read-only restriction.
```

:::

### Group access and permissions

The **Access & Permissions** section of a group is where you can assign users the right level of access based on their role or responsibilities. You decide:

> 

Author: ```suggestion
```
